ee/server/service/scep_proxy.go

```diff
@@ -0,0 +1,208 @@
+package service
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"regexp"
+	"strings"
+
+	"github.com/Azure/go-ntlmssp"
+	"github.com/fleetdm/fleet/v4/pkg/fleethttp"
+	"github.com/fleetdm/fleet/v4/server/contexts/ctxerr"
+	"github.com/fleetdm/fleet/v4/server/fleet"
+	scepclient "github.com/fleetdm/fleet/v4/server/mdm/scep/client"
+	scepserver "github.com/fleetdm/fleet/v4/server/mdm/scep/server"
+	"github.com/go-kit/log"
+	"golang.org/x/text/encoding/unicode"
+	"golang.org/x/text/transform"
+)
+
+var _ scepserver.ServiceWithIdentifier = (*scepProxyService)(nil)
+var challengeRegex = regexp.MustCompile(`(?i)The enrollment challenge password is: <B> (?P<password>\S*)`)
+
+const (
+	fullPasswordCache             = "The password cache is full."
+	MessageSCEPProxyNotConfigured = "SCEP proxy is not configured"
+)
+
+type scepProxyService struct {
+	ds fleet.Datastore
+	// info logging is implemented in the service middleware layer.
+	debugLogger log.Logger
+}
+
+// GetCACaps returns a list of SCEP options which are supported by the server.
+// It is a pass-through call to the SCEP server.
+func (svc *scepProxyService) GetCACaps(ctx context.Context) ([]byte, error) {
+	appConfig, err := svc.ds.AppConfig(ctx)
+	if err != nil {
+		return nil, ctxerr.Wrap(ctx, err, "getting app config")
+	}
+	if !appConfig.Integrations.NDESSCEPProxy.Valid {
+		// Return error that implements kithttp.StatusCoder interface
+		return nil, &scepserver.BadRequestError{Message: MessageSCEPProxyNotConfigured}
+	}
+	client, err := scepclient.New(appConfig.Integrations.NDESSCEPProxy.Value.URL, svc.debugLogger)
+	if err != nil {
+		return nil, ctxerr.Wrap(ctx, err, "creating SCEP client")
+	}
+	res, err := client.GetCACaps(ctx)
+	if err != nil {
+		return res, ctxerr.Wrap(ctx, err,
+			fmt.Sprintf("Could not GetCACaps from SCEP server %s", appConfig.Integrations.NDESSCEPProxy.Value.URL))
+	}
```

Comment:
nit: here and in other occurrences below, you can do `ctxerr.Wrapf`

```suggestion
		return res, ctxerr.Wrapf(ctx, err, "Could not GetCACaps from SCEP server %s", appConfig.Integrations.NDESSCEPProxy.Value.URL)
	}
```